File: typescript/typelayer-common/wire.ts
// Module: wire
import * as typelayerCommonTypes from "./types";


export class DecoderException extends  typelayerCommonTypes.TypelayerException {
    offset: string;
    constructor(m: string, meta: {offset: string;}) {
        super(m, "typelayer_common_wire", "decoder_exception", new Map([["offset", meta.offset]]));
        Object.setPrototypeOf(this, DecoderException.prototype);
        this.offset = meta.offset;
    }
}

File: python/typelayer_common/typelayer_common/wire.py
from . import types as typelayer_common_types
from dataclasses import dataclass
from typing import List, Dict, Optional


class DecoderException(typelayer_common_types.TypelayerException):
    offset: str;
    def __init__(self, m: str, offset: str):
        super().__init__(m, "typelayer_common_wire", "decoder_exception", {"offset": offset})
        self.offset = offset;